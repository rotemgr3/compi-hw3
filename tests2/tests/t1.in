//piazza 64 - should be ok +
override int foo(int a) {return 5;}
override int foo(int a, int y) {return 5;}
override bool foo(int a) {return true;}
override bool foo(byte a) {return true;}
override int foo(byte a) {return 5;}
override int foo(int a, bool y) {return 5;}
override int foo(bool a, int y) {return 5;}
void main() {return;}
