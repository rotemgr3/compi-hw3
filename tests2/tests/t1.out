---end scope---
a INT -1
---end scope---
a INT -1
y INT -2
---end scope---
a INT -1
---end scope---
a BYTE -1
---end scope---
a BYTE -1
---end scope---
a INT -1
y BOOL -2
---end scope---
a BOOL -1
y INT -2
---end scope---
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
foo (INT)->INT 0
foo (INT,INT)->INT 0
foo (INT)->BOOL 0
foo (BYTE)->BOOL 0
foo (BYTE)->INT 0
foo (INT,BOOL)->INT 0
foo (BOOL,INT)->INT 0
main ()->VOID 0
